lib/bootsnap/load_path_cache/path_scanner.rb

```diff
@@ -1,8 +1,11 @@
+require_relative '../explicit_require'
+
+Bootsnap::ExplicitRequire.from_rubylibdir('set')
+
 module Bootsnap
   module LoadPathCache
     module PathScanner
-      REQUIRABLES_AND_DIRS = "/**/*{#{DOT_RB},#{DL_EXTENSIONS.join(',')},/}"
-      IS_DIR = %r{(.*)/\z}
+      REQUIRABLES_AND_DIRS = "/**{,*/**}/*{#{DOT_RB},#{DL_EXTENSIONS.join(',')},/}"
```

Comment:
So expanding the first curlies, you get:

```ruby
"/**/*{#{DOT_RB},#{DL_EXTENSIONS.join(',')},/}"
"/***/**/*{#{DOT_RB},#{DL_EXTENSIONS.join(',')},/}"
```

Am I missing something? I've never seen a triple asterisk before and I can't find any references for it. I don't understand why this change would make it resolve symlinks. I'm probably just missing some weird bit of arcana on how glob works?

At any rate, this pattern was already verging on too complex for one line and no documentation, I think it will probably need to be written more clearly somehow if we're going to add more complexity to it like this.

Reply:
That's my bad. I have missed one `/` and it for some reason worked ¯\_(ツ)_/¯.  I have found better glob pattern to find everything we need without duplicates (I can get back to arrays instead of sets).